### ISSUE TITLE
Why are versions "pinned" with Craft updates?

### ISSUE THREAD

**Issue by frank-laemmer** [craftcms/cms#5727]
TLDR; Why `3.5.5` and not `^3.5.5` in `composer.json`?

### Background

We (fortrabbit) have always suggested our clients to just use `composer update` (on the local machine) to update Craft CMS as well as dependencies and plugins for a long time.

Just recently we found out that, when installing or updating from the Craft Control Panel or the `./craft update` CLI command the `composer.json` file will contain "pinned" versions. 

Pinned with Craft update mechanism:

```
"craftcms/cms": "3.5.5"
```

Expected from using Composer:

```
"craftcms/cms": "^3.5.5"
```

Initial installation with Composer will not pin the versions. Versions get pinned after using the Control Panel or the Craft CLI command for the first time. After that, `composer update` will not have the desired effect (updating minor and patch versions), since versions will stick to the pinned version. To get back to composer driven updates, one has to modify composer json by inserting the Caret in front.

### Questions

* Why do you pin the versions?
* Is it maybe possible to not pin the versions to be more Composer compatible?

### Side remarks

The same mechanism applies to plugins. Once plugins are installed or updated via the plugin store / Craft Control Panel, they are pinned. 

I see, when using Composer to update, the Control Panel is not showing me the previously installed versions with the changelog, it just knows that versions are up-to-date. When updates are done via Craft, the Control Panel shows the installed versions.

I usually prefer `composer update` over `./craft update` in terminal since it gives me a more verbose output of the packages that got updated to which version.

I have not examined how 3rd party dependencies, such as `vlucas/phpdotenv` are handled using the different methods.

### Additional info

- Craft version: 3.x.x
- PHP version: N/A
- Database driver & version: N/A
- Plugins & versions: N/A

**Comment by thijskaspers** [comment]
I think this is done to force people to look at the changelog? Blindly running `composer update` could introduce breaking changes and mess up your site without you knowing. When you run `./craft update` or look at the update screen in the control panel, at least you're confronted with important info/warnings/notices regarding the available updates. Sometimes there are additional steps required before or after an update to keep everything running smoothly.

Also, in edge cases, Craft has to update itself to a certain version before being able to update to an even newer version.

Just my 2 cents. Someone from P&T team will probably have a better answer to this.

**Comment by brandonkelly** [comment]
Two reasons:

1. If we set the requirement to `^1.2.3` (for example), that introduces a race condition where the actual version you end up with could be different than the one you thought you were getting, per the Updates utility, if a new version was released between the time you landed on the utility and the time you click “Update”. While that’s a possibility when running `composer update` with a lose dependency requirement as well, at least `composer update` won’t run DB migrations automatically after it’s done, so even if you end up with an unexpected version, you still have a chance to roll back to a prior version before running `./craft migrate/all`.

2. Craft and any commercial plugins have expirable updates. If you update to v1.2.3 within your window, but the Composer requirement is actually set to `^1.2.3`, and then your license expires, and then you run `composer update` manually, you risk being updated to a version that you are not licensed to use. If the update came with any new DB migrations, then you won’t be able to (easily) roll back to v1.2.3 (your maximum licensed version), and will be forced to renew your license when you weren’t expecting to. This is a risk if you manually add the `^`, but at least you know you did that.

**Comment by frank-laemmer** [comment]
Thanks for explanations!
